File: frontend/src/app/dtos/jobRequest.ts
export interface JobRequest {
  id?: number;
  title: string;
  description?: string;
  category: string;
  status: string;
  deadline: string;
  createdAt?: string;
  customerId?: number;
  propertyId?: number;
  displayImageUrl?: string;
  lowestPrice?: number;
}

export interface JobRequestSearch {
  id?: number;
  title?: string;
  description?: string;
  category?: string;
  status?: string;
  deadline?: string;
  propertyId?: string;
  distance?: number
  lowestPriceMin?: number;
  lowestPriceMax?: number;
}

export interface JobRequestUpdate {
  title: string;
  description?: string;
  category: string;
  status?: string;
  deadline: string;
  propertyId?: number;
}

export interface JobRequestCreateEdit {
  propertyId?: number;
  title: string;
  description?: string;
  category: string;
  status?: string;
  deadline: string;
  jobRequestImages?: JobRequestImage[];
}

export interface JobRequestImage {
  id: number;
  imageType: string;
  displayPosition: number;
  downloadUrl: string;
}

export interface JobRequestPrice {
  id: number;
  title: string;
  description?: string;
  category: string;
  status: string;
  deadline: string;
  lowestPrice: number;
  address: string;
}

export interface JobRequestWithUser {
  id?: number;
  title: string;
  description?: string;
  category: string;
  status: string;
  deadline: string;
  createdAt?: string;
  customerId?: number;
  propertyId?: number;
  customerUsername: string;
}

File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/entity/JobRequest.java
package at.ac.tuwien.sepr.groupphase.backend.entity;

import at.ac.tuwien.sepr.groupphase.backend.type.Category;
import at.ac.tuwien.sepr.groupphase.backend.type.JobStatus;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OrderBy;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * Represents a job request in the persistent data store.
 */
@Entity
@Table(name = "job_request")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(exclude = {"jobRequestImages", "receivedJobOffers"})
@EqualsAndHashCode(exclude = {"jobRequestImages", "receivedJobOffers"})
public class JobRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private ApplicationUser customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "property_id")
    private Property property;

    private String title;

    @Column(length = 4095)
    private String description;

    @Enumerated(EnumType.STRING)
    private Category category;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private JobStatus status = JobStatus.PENDING;

    private LocalDate deadline;

    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "jobRequest")
    @OrderBy("displayPosition DESC")
    private List<JobRequestImage> jobRequestImages;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "jobRequest")
    @OrderBy("createdAt DESC")
    private List<JobOffer> receivedJobOffers;

}